packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,220 @@
+# Checkbox Decisions <!-- omit in toc -->
+
+- [Design](#design)
+- [Anatomy](#anatomy)
+- [`Checkbox` API](#checkbox-api)
+  - [Examples:](#examples)
+    - [Basic](#basic)
+    - [Validations](#validations)
+- [`CheckboxGroup` API](#checkboxgroup-api)
+  - [Examples:](#examples-1)
+    - [Basic](#basic-1)
+    - [Controlled, Uncontrolled](#controlled-uncontrolled)
+  - [Note on CheckboxGroup](#note-on-checkboxgroup)
+- [Component Architecture](#component-architecture)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=13227%3A163026) to all variants of the Checkbox component
+
+## `Checkbox` API
+
+| Prop                | Type       | Required | Default   | Description                                                                                                                                              |
+| ------------------- | ---------- | -------- | --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| children            | `string`   | No       | undefined | The text to be rendered as CheckboxLabel                                                                                                                 |
+| helpText          | `string`   | No       | undefined | The helper text to be rendered                                                                                                                           |
+| isChecked           | `boolean`  | No       | false     | state of the checkbox, if explicitly set the checkbox will be controlled                                                                                 |
+| defaultChecked      | `boolean`  | No       | false     | default state of the checkbox, if isChecked is not provided the checkbox will be uncontrolled                                                            |
+| onChange            | `Function` | No       | undefined | The function to be called when the checkbox state changes.                                                                                               |
+| isIndeterminate     | `boolean`  | No       | false     | mixed state of the checkbox                                                                                                                              |
```

Comment:
No doesn't depend on isChecked. isIndeterminate will override the isChecked state. 

So if you have 
<Checkbox isChecked={true} isIndeterminate={true} />

The checkbox will be in indeterminate state and because we don't want. 
example: https://github.com/razorpay/blade/pull/544/files#diff-ecad1a0cb9d580749fedae8066d59e12d65733f666f0f39cc9389af1bfd057ffR129